### PR TITLE
added an option to force the file output format when using OUTPUT_BY_…

### LINKED ISSUE #645: RevertSam and OUTPUT_BY_READGROUP does not allow specify output file format
When using RevertSam and OUTPUT_BY_READGROUP  the output file format is depend on the input file format (O= should be a directory).

Currently it is impossible to select the output format with this tool. It would be nice if there is an option that selects the sam, bam or cram file format as output. 

A small discussion on this topic can be found at:
http://gatkforums.broadinstitute.org/gatk/discussion/comment/32736#Comment_32736

I hope someone is will to work on this feature request. My Java is a bit too rusty to implement this myself, but it does not look to me like a hard thing to implement (I did took a peek into the code).

### PR DESCRIPTION
**Pull request by maarten-k**
…READGROUP  which should fix #645 .

Comments are more then welcome since I am not sure this is the best way to implement this feature.

### CODE REVIEW

File: src/main/java/picard/sam/RevertSam.java [hunk 1]
```diff
@@ -121,6 +122,11 @@
     @Option(shortName = "OBR", doc = "When true, outputs each read group in a separate file.")
     public boolean OUTPUT_BY_READGROUP = false;
 
+    public static enum FileType {sam, bam, cram,dynamic}
+    @Option(shortName = "OBRFF", doc = "When using OUTPUT_BY_READGROUP, the output file format can be set to a certain format." )
+    public FileType  OUTPUT_BY_READGROUP_FILE_FORMAT=FileType.dynamic;
+
+
```
Comment (nh13) [inline src/main/java/picard/sam/RevertSam.java]: extra nl

File: src/main/java/picard/sam/RevertSam.java [hunk 2]
```diff
@@ -208,8 +214,15 @@ protected int doWork() {
...
-        if (OUTPUT_BY_READGROUP) {
-            final String defaultExtension = getDefaultExtension(INPUT.toString());
+        if (OUTPUT_BY_READGROUP){
+
+            final String defaultExtension;
+            if (OUTPUT_BY_READGROUP_FILE_FORMAT==FileType.dynamic){
+                defaultExtension = getDefaultExtension(INPUT.toString());
+	    }else{
+                defaultExtension="."+OUTPUT_BY_READGROUP_FILE_FORMAT.toString();
+	    }
```
Comment (nh13) [inline src/main/java/picard/sam/RevertSam.java]: formatting in lines 220-224

**Comment by nh13** [comment]
The code formatting needs some help, but after that 👍

**Comment by maarten-k** [comment]
@nh13 I improved the code formatting.

**Comment by yfarjoun** [comment]
@nh13 gave it a 👍 upon code formatting. The code has been formatted, so I'm merging.

**Pull request merged**